docs-2.0/4.deployment-and-installation/5.zone.md

```diff
@@ -1,67 +1,85 @@
+
+Since each Zone contains a complete set of graph space data partitions, at least one Storage node is required within each Zone to store these data partitions.
+
+The partition replicas in NebulaGraph achieve strong consistency through the [Raft](../1.introduction/3.nebula-graph-architecture/4.storage-service.md#raft_1) protocol. It's recommended to use an odd number of partition replicas, and therefore, it's also suggested to set an odd number of Zones.
+
+Taking the following picture as an example, when creating a graph space (S1), the data is partitioned into 3 partitions, with 3 replicas for each partition, and the number of Zones is also 3. Six machines hosting the Storage service are paired up and added to these 3 Zones. When creating the graph space, NebulaGraph stores the 3 replicas of each partition evenly across zone1, zone2, and zone3, and each Zone contains a complete set of graph space data partitions (Part1, Part2, and Part3).
+
+<img src="https://docs-cdn.nebula-graph.com.cn/figures/core_zone_230913.png" width="800" alt="example_for_zones">
 
-![Zone](https://docs-cdn.nebula-graph.com.cn/figures/zone1.png)
+To reduce cost of cross-Zone network traffic, and increase data transfer speed (Intra-zone network usually has a lower latency than inter-zone network), you can configure the Graph service to prioritize intra zone data access. Each Graphd will then prioritize to access the partition replica in the same zone as specified by Graphd if there is any. As an example, suppose Graph A and Graph B are located in zone1, Graph C and Graph D are in zone2, and Graph E is in zone3. You can configure Graph A and Graph B to prioritize accessing data in zone1, Graph C and Graph D to prioritize accessing data in zone2, and Graph E to prioritize accessing data in zone3. This helps reduce the cost of cross-zone network traffic and improves data transfer speed.
 
-In the above figure, the six machines with the running storage service are grouped in pairs and added to three Zones. When creating a graph space S1 with a partition replica number of 3 in these three Zones, these partition replicas are evenly stored in Zone1 ~ Zone3.
+<img src="https://docs-cdn.nebula-graph.com.cn/figures/core_intra_zone_20230913.png" width="800" alt="example_for_intra_zone">
+
+In cases where data retrieval fails due to a Storage node failure in a specific zone, you can configure the Graph service to read leader replica data across zones to ensure data availability. For instance, if machine B experiences a failure, causing Graph A to be unable to read the Follower replica data of Part3 in zone1, you can configure Graph A to read the Leader replica data of Part3 in zone3, thus ensuring data availability.
```

Comment:
Hold  on a second. This may not be the case at this moment. Let me confirm.

Reply:
Let's delete this paragraph. It's the future feature. Currently on storaged failure, we will get 'E_RPC_FAILURE' from the response of a storaged request. Then we will either return this code or ignore the code if `accept_partial_success` is set to true. In the future, after we implement the retry functionality, we will have ways to retry request to other replicas (not necessarily leader replica). In the current implementation, we do not do any retry. We just determine whether send request to leader or the replica in the same zone in advance. We only send request to leader in the following cases:
```
When intra zone reading is enabled, if zone is not given or the zone is empty, or there is no part host in this zone (impossible under the constraint of #replicas = # zones) && allow rerouting, fallback to leader. If intra zone reading is
disabled, fall back to leader. If it's write request, fall back to leader.
```